question21/main.js

```diff
@@ -0,0 +1,159 @@
+'use strict';
+const wrapper = document.getElementById('js-wrapper');
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const showLoadImg = () => {
+  const loadImg = createElementWithClassName("img", "loading");
+  loadImg.src = "./img/loading-circle.gif";
+  loadImg.id = 'js-loading';
+  wrapper.appendChild(loadImg);
+} 
+
+const removeLoadImg = () => {
+  document.getElementById('js-loading').remove();
+} 
+
+const getData = new Promise((resolve) => {
+  showLoadImg();
+  setTimeout(() => {
+    resolve(fetch('https://mocki.io/v1/5381547b-04ff-4a89-b169-dd84425f5196'));
+  }, 3000);
+})
+
+//Get json data
+const callApi = async() => {
+  try {
+    const response = await getData;
+    if (!response.ok) {
+      console.error(`${response.status}:${response.statusText}`);
+    }
+    const json = await response.json();
+    return json.data;
+  }
+  catch(error) {
+    console.error(error);
+  }
+  finally {
+    removeLoadImg();
+  }
+}
+
+const init = async() => {
+  const userData = await callApi();
+  if (userData){
+    renderTabaleElement(userData);
+    } else {
+    wrapper.textContent = "データの表示に失敗しました";
+    console.error('データの表示に失敗しました'); 
+  }
+}
+init();
+
+const userDataColumn = {
+  userId: "ID",
+  name: "名前",
+  gender: "性別",
+  age: "年齢"
+}
+
+const renderTabaleElement = (userData)  => {
```

Comment:
nits 
```suggestion
const renderTableElement = (userData)  => {
```

Reply:
Thank you for pointing it our! Fixed.
fixed commit : https://github.com/csakyo/homework/pull/36/commits/838ac36e03e49f4b82c4ded1b9e2392cf9dbf724